### PR TITLE
Adding GraphQL::Generator tests

### LINKED ISSUE #92: Need specs for GraphQL base_query
Follow up to #90 

cc @abellotti

### PR DESCRIPTION
**Pull request by abellotti**
Adding GraphQL::Generator test for the schema_overlay logic to test base_query

Fixes #92

### CODE REVIEW

**Comment by abellotti** [comment]
Keeping as WIP for now, depending on order, other tests unreliability fail.

**Comment by abellotti** [comment]
The random spec failures were constant problems.  Moved the dynamically generated GraphQL schema away from ::Api into its own ::ManageIQGraphQLApi.

File: lib/manageiq/api/common/graphql.rb [hunk 1]
```diff
@@ -9,10 +9,15 @@
 require "manageiq/api/common/graphql/types/date_time"
 require "manageiq/api/common/graphql/types/query_filter"
 
+module ManageIQGraphQLApi
```
Comment (bdunne) [inline lib/manageiq/api/common/graphql.rb]: I don't like that this constant is not in the `ManageIQ::Api::Common` namespace

**Comment by abellotti** [comment]
Does this fit the bill ?

```
::ManageIQ::API::Common::GraphQL::Api
```

**Pull request merged**